src/structure/pdb.jl

```diff
@@ -30,26 +39,231 @@ end
 
 
 """
-Download a Protein Data Bank (PDB) file or biological assembly from the RCSB
-PDB. By default downloads the PDB file; if `ba_number` is set the biological
-assembly with that number will be downloaded.
+Returns a list of all PDB entries from RCSB PDB server
 """
-function downloadpdb(pdbid::AbstractString,
-                    out_filepath::AbstractString="$pdbid.pdb";
-                    ba_number::Integer=0)
+function getallpdbentries()
+    pdbidlist = Array{String,1}()
+    info("Fetching list of all PDB Entries from RCSB PDB Server...")
+    download("ftp://ftp.wwpdb.org/pub/pdb/derived_data/index/entries.idx","entries.idx")
+    open("entries.idx") do input
+        # Skips the first two lines as it contains headers
+        linecount = 1
+        for line in eachline(input)
+            if linecount > 2
+                # The first 4 characters in the line is the PDB ID
+                push!(pdbidlist,line[1:4])
+            end
+            linecount +=1
+        end
+    end
+    rm("entries.idx")
+    return pdbidlist
+end
+
+
+"""
+Returns a list of pdb codes in the weekly pdb status file from the given URL. 
+"""
+function getstatuslist(url::AbstractString)
```

Comment:
The URL is supposed be weekly status file from RCSB Server. I will add it in the docstrings.